app/javascript/admin/controllers/config_controller.js

```diff
@@ -245,4 +259,64 @@ export default class ConfigController extends Controller {
     }
   }
   // AUTH PROVIDERS FUNCTIONS END
+
+  enabledProvidersWithMissingKeys() {
+    const providersWithMissingKeys = [];
+    document
+      .querySelectorAll('[data-enable-auth="true"]')
+      .forEach((provider) => {
+        const providerName = provider.dataset.authProviderEnable;
+        const officialName = provider.dataset.authProviderOfficial;
+        if (
+          !document.getElementById(`site_config_${providerName}_key`).value ||
+          !document.getElementById(`site_config_${providerName}_secret`).value
+        ) {
+          providersWithMissingKeys.push(officialName);
+        }
+      });
+
+    return providersWithMissingKeys;
+  }
+
+  generateProvidersList(providers) {
+    let list = '';
+    providers.forEach((provider) => {
+      list += `<li>${provider}</li>`;
+    });
+    return list;
```

Comment:
nitpick/refactor (non-blocking): An array reduce would work well here.

```suggestion
    const list = providers.reduce((html, provider) => {
      return `${html}<li>${provider}</li>`;
    }, '')
   
    return list;
```

Reply:
+1 +1 to using `reduce` here!! 🔥

Reply:
TIL! 🙏🏾  Thanks @nickytonline